src/app/utility/meter-grouping/meter-grouping.component.ts

```diff
@@ -201,4 +203,18 @@ export class MeterGroupingComponent implements OnInit {
     this.utilityMeterGroupDbService.deleteIndex(this.groupToDelete.id);
     this.closeDeleteGroup();
   }
+
+  setToggleView(typeID,groupId) {
```

Comment:
in here you can just pass the meterGroup object.

then just do:

`meterGroup.visible = !meterGroup.visible`
`update(meterGroup)`


The arrays are updated after the group get's updated in the db.